### PR TITLE
chore: Fixed IQA failed issue #27650

### LINKED ISSUE #27650: Edit Page V2: Contentlet inserts in wrong position when using contentlet tools
### Parent Issue

https://github.com/dotCMS/core/issues/27546

### Problem Statement

When using contentlet tools (Blue plus button that appears on top and down of a contentlet), we can expect that any content type added with these buttons can be inserted in a wrong position. This can be a really weird behavior for the end user, because the contentlet will appear anywhere but the position the user selected.

### Steps to Reproduce

1. Open new Edit Page
2. Hover in a contentlet
3. Click in one of the blue plus buttons that appeared (A menu should appear)
4. Hover on another contentlet, make sure that the blue border and buttons appeared on the contentlet you hovered
5. Try to click on an option in the menu without hovering in another contentlet
6. Select a contentlet that is not duplicated
7. See how the contentlet was added in a different place that you selected in the 3rd step

## Screenshot following the steps and adding another cases

https://github.com/dotCMS/core/assets/63567962/979d725b-3135-44fe-9b83-e45930b4bd43

### Acceptance Criteria

* User should be able to add a contentlet in a selected position without problems

### dotCMS Version

`master-latest`

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### Insights

We have a problem with the flow of how this works.

This flow is not a staged process but a parallel one. This means that we can reset the contentlet we first selected as position, while still choosing which type of content are we going to add. This ends in this weird behavior, where we didn't selected any type of content but can still make hover in another contentlet and see the buttons and also click on them, resulting in a contentlet added in a wrong position if we make the correct steps.

In technical terms, when we hover on a contentlet the client sends a post message that tells the editor to set the new contentlet as the pivot for the position of the new contentlet (works as reference to add before or after of it ). And this post message is send even if we already selected a contentlet as pivot to add the new contentlet (click on the blue button).

Here are some Excalidraws to see the current flow and how it could be.

#### Current flow
https://excalidraw.com/#json=09ubKwCp97QNmphkp7gHA,8W1DTltNK4kQemabPXuvGg

#### Proposed Flow
https://excalidraw.com/#json=JXO8t9LJJGBEj469sp_xd,HnrijBbBI1E-UX5ubpHo0Q

### Assumptions & Initiation Needs

* We understand the event flow of Edit Page V2
* We understand the impact in a UX point of view

### Quality Assurance Notes & Workarounds

I would recommend to QA, that we need to test all the behaviors in all contentlets and containers, empty or not. We found this bug while testing the behavior of the add using contentlets that the client didn't have components for (No component for XYZ).

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Proposed Changes
* Fixed reported IQA 

When the user change to hover to the same contentlet but in different container, the menus are hidden

https://github.com/dotCMS/core/assets/144152756/54277ff8-2ff5-4e22-b7b9-9e264ff09fe2

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 1]
```diff
@@ -101,7 +101,9 @@ export class EmaContentletToolsComponent implements OnInit, OnChanges {
 
         if (
             changes.contentlet.currentValue?.payload.contentlet.identifier !==
-            changes.contentlet.previousValue?.payload.contentlet.identifier
+                changes.contentlet.previousValue?.payload.contentlet.identifier ||
```
Comment (oidacra) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:104]:
```
if (this.hasContentletOrContainerChanged(changes.contentlet)) {
        this.hideMenus();
}
```

```
private hasContentletOrContainerChanged(contentletChange: SimpleChange): boolean {
    const currentValue = contentletChange.currentValue?.payload;
    const previousValue = contentletChange.previousValue?.payload;

    const hasIdentifierChanged = currentValue.contentlet.identifier !== previousValue.contentlet.identifier;
    const hasUUIDChanged = currentValue.container.uuid !== previousValue.container.uuid;

    return hasIdentifierChanged || hasUUIDChanged;
}
```

It is more explicit what means that.

Comment (KevinDavilaDotCMS) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts]: Totally agree
Changed
Thanks!!

**Review by oidacra — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 2]
```diff
@@ -213,6 +211,25 @@ export class EmaContentletToolsComponent implements OnInit, OnChanges {
...
+        const currentValue = contentletChange.currentValue?.payload;
+        const previousValue = contentletChange.previousValue?.payload;
+
+        const hasIdentifierChanged =
+            currentValue?.contentlet.identifier !== previousValue?.contentlet.identifier;
+        const hasUUIDChanged = currentValue?.container.uuid !== previousValue?.container.uuid;
+
+        return hasIdentifierChanged || hasUUIDChanged;
+    }
+
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:232]: I don't think is responsibility of the contentlet to evaluate if something change here... I actually had the same problem and I solve it in the store: https://github.com/dotCMS/core/blob/49f0621cb4b492c5d10403a0e009aaa6eacb705c/core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/store/dot-ema.store.ts#L116-L131
